### ISSUE TITLE
EU4Vic2 1.22 update

### ISSUE THREAD

**Issue by derHochmeister** [Idhrendur/paradoxGameConverters#291]
I don't know how to upload and pull request, guys, sorry.

I put the update here, so that @kratostatic, who has became the leader of the EU4Vic2 project (I think), can see it. They are to put in `/EU4toV2/Data_Files`.

Link: http://www.mediafire.com/file/1piwj41fxa71twq/EU4-1.22.rar

Changelog:

- Country  mappings reordered and fixed to avoid duplicates (as promised)
-- Now any country mapping has the country it refers to named, to avoid confusion
-- Fixed so the Jangladesh=>Jan Mayen mapping. Jangladesh now maps to the historical Bikaner State
-- Discovered and fixed the Zhou=>Wu mapping
-- Fixed the various lines mixing Qing, Qin and China
-- Bettered the legend at the start to make it comprehensible for modding (some countries are assigned CK2 titles, e.g. León, even if they have an EU4 tag and, not having a line, may seem to have been forgotten)
-- Added all the countries that were actually forgotten. Flags are taken from historical and posthumous symbols
-- Discovered and fixed the Kochin=>Venad and Venad=>Venad mapping (now Kochin maps to Kochin and Venad to historical Travancore)
-- Added some countries that may be created as client/custom/mod and still have equivalent in Vic2
- Fixed some CK2 flags to be senseful
-- Hospitaller flags taken from KNI
-- Templar flags based on attributed Templar flag and fantasy
-- Beloozero flags based on Principality (def, mon), current city (rep), fantasy (others)
-- Rostov flags based on city's flag and historical coats. Communist and fascist are fantasy.
-- Fixed Maldives flag to be historical and not eyerapish.
- Added some countries not basically in EU4 or CK2 but that are in Vic2 (e.g. Belgium, Ecuador)
- Recontrolled culture mappings. Added conversion to Central American, Manipuri, Maures.
- Added Métis (any native North American with christian religion)
- Revised culture conversion to Neo-European and American cultures
- fixed postcolonial decision sometimes not fired
- fixed annex SoIs event to not annex released colonies
- added CSA and Taiping to the postcolonial decision to not screw up events
- fixed Boma, 1.12~1.15 province conversions missing
- cultural cores re-fixed (it is the same as https://github.com/Idhrendur/paradoxGameConverters/pull/286)
- Babylonian now maps to Assyrian culture (from the CK2 632 mod conversion)

**1.22 features:**

-Tsardom maps to Absolute Monarchy, Veche Rep. to Democracy, Principality to HM's Government.
-Rostov and Beloozero are already mapped due to their CK2 titles.
-New provinces of Russia and Ruthenia have been added.

**Comment by derHochmeister** [comment]
The countries that may be in EU4 through mods/clients that are supported are: Texas, Belgium, Uruguay, Bolivia, Ecuador, South Africa, Philippines, New Zealand, Yugoslavia, Byelorussia, Two Sicilies, Austria-Hungary and Czechoslovakia. Also, Azerbaijan, Schleswig and Mongolia, exportable from CK2, are converted to the Victorian equivalent.

**Comment by kratostatic** [comment]
Awesome, that's great!  I'll be adding this in over the next few days - I'll say here if there are any issues.

**Comment by kratostatic** [comment]
OK so most of this is now in.  Unfortunately there seems to be an issue with the converter decisions part - Victoria 2 crashes on loading whenever I include it.  I'm going to be working on a potential fix, but in the meantime is there any way you can think getting the CSA and Taiping in other than using the postcolonial decision?  That's the only change compared to the last version that i can see.

**Comment by derHochmeister** [comment]
It doesn't crash to me, this is weird.

**Comment by derHochmeister** [comment]
Have you completely pasted the file? Because a little change in the line order can make it crash.

**Comment by kratostatic** [comment]
I thought I had but I just tried again with a fresh download and it works!

I copied the file into one of the test games and there are a bunch of very strangle things going on.  There's a very small dutch cultured USA in the game which suddenly has cores over the vanilla USA territory for no reason.  Also, all the American countries have their vanilla cores and if they are released then they have their original cultures (most of which don't even exist in a converted game so they have no primary culture pops at all).  This makes even less sense in this case when Brazil is owned by Spain, Argentina and Chile by Brittany and a dutch cultured Argentina and most of the rest of the Americas by the Livonian Order!  On checking with another game the same kind of things seem to be happening there.

Is there a quick way to modify the decision to have it make a bit more sense?

**Comment by derHochmeister** [comment]
The solution is to change which states get cored by which countries. But you can't deny that all New World post-colonial countries do deserve their lands. I mean, the US, regardless of language, would have claimed up to Chicago for sure.

**Comment by kratostatic** [comment]
TL;DR - I'm not keeping this decision as is or close to it, I'll change it to only add CSA and Taiping if there isn't a workaround, and have the rest of it be an optional decision.

Yes I can deny that the New World post colonial countries deserve their lands - why would they even exist at all in a converted game?  The circumstances which led to the 1836 set-up of countries rely on a specific series of events - colonisation by Spain, Portugal, the UK and France, how they choose to divide up the continent between themselves, how the various colonial conflicts go, the US war of independence, then Napoleon screwing up Spain and Portugal causing their colonies to self govern and selling the US more land than it already owned, and finally the whole Simon Bolivar thing.  None of these things have to happen in EU4 - most of them are actually pretty difficult to have happen.  Change any one and the vanilla 1836 set-up makes no sense at all in terms of borders, nor the nations themselves, nor the language they speak - the vanilla Vicky 2 post-colonial cultures won't even exist in most games.  Hell, Spanish, Portuguese, English and French might not exist in some games let alone the associated post-colonial cultures.

And why would a country which happens to be called the United States of America in this alternate history have exactly the same expansionist policies as our US?  It's literally just a name of a country that any EU4 colonial nation in the right area can form on independence.  It doesn't have to have anything else in common at all - it can be a monarchy, it can be an OPM or own literally the entire continent (this dutch US was pretty small - owned two and a bit states I think?), it can have any primary or accepted culture including native American culture(s).  

Remember also that there will be "colonial nations" created by the converter from the colonies in EU4 so why create new ones?  I agree that the way these are implemented can result in strangeness once converted - as you know  from the way forming cultural unions used to merge in colonial nations before you fixed that decision.  I have thoughts on solutions - I think a couple are referenced in issues on here - and I'll be working on it once I've got this current version sorted.

Anyway in the last version of the converter I moved this decision into a different folder called "optional decisions" and changed the decision set-up so that rather than it being taken by the AI it would be taken by the player.  The mod doesn't see the decision unless the user actively moves it into the decision folder. Now I'm happy to leave it there and change it so that it would be done automatically, or even have a decision that only the player can take as part of the default decisions.  I'll create a separate decision to create cores for the CSA and Taiping as those are causing actual bugs for the converted game.  But I'm not releasing a version of the converter which creates a bunch of nations, cultures and cores out of absolutely nowhere as soon as the game starts with no input from the player.

Sorry for the rant, and I know you've spent a lot of time on this but I really disagree with this entire concept.

**Comment by derHochmeister** [comment]
I understand your motivations, to be frank. But Canada should still be able
to be formed if a nation in the area is generated. To be fair, the names
are the problem but, let's face it, the Byzantine Empire was never called
so in EU timeframe, yet none actually said "hold on, this ahistorical". I
think you should add the player-based decision, but letting it in the
decisions folder, so that the player actually decides, since most may not
even be aware of such decision's existance.

**Comment by kratostatic** [comment]
Yeah I agree on giving the player choice about this and good point that they might not see the extra folder.  I'll merge in the version you've pulled on my branch.

(In most of my converted games the colonial nations formed from the colonising powers in EU4 get released as dominions pretty quickly even if liberty threshold is set to 100.  These often go independent either through being released in a war or in a couple of cases becoming great powers so I think that they won't normally be forever linked to their coloniser anyway, but this way would make sure - sometimes they never get released or don't exist in the first place if the coloniser only colonised 4 provinces in a region.)

**Comment by kratostatic** [comment]
Although shouldn't the colonial nations from EU4 do everything the nations created by this decision do?  Their cores exist nearly all the time and depending on the absorption threshold chosen they can even exist as dominions at game start.  So they're no different to say Vanilla Vicky2 Canada or Australia, and the others went independent before 1836 so would appear as independent nations anyway.

I'll check with an 1821 start EU4 save in a bit, hang on.

**Comment by derHochmeister** [comment]
The buggy thing about exported CNs is they'll forever be "Xese Ya", while
the postcolonial cores ensure they won't. That's why I included the ability
to form US or Canada by core if a CN.

**Comment by kratostatic** [comment]
So anyway, all of this is in and unless I've missed something the converter already works for 1.22 saves.

**Comment by kratostatic** [comment]
So I was just doing final testing before release and unfortunately the nation forming decisions have gone catastrophically wrong somewhere.  Rather than the country forming decisions being visible, they create a couple of decisions called things like "potential" or "exists" which every nation can take (and which every nation does take every day) and which do absolutely nothing.  I've uploaded the test save I'm using here as I'm out of ideas as to how to fix this.

[1_22_test_save.zip](https://github.com/Idhrendur/paradoxGameConverters/files/1131143/1_22_test_save.zip)

By the way I've been using Scottish Columbia to test as I think that has all the cores required to form Columbia, or only needs one of the uncolonised cores next door if not.

**Comment by kratostatic** [comment]
Oh and just to say if we can get this working it will be a lot more elegant than what I thought it was at first, so nice work!

**Comment by derHochmeister** [comment]
Sadly, the bug of "allow_title" and similar is a thing that keeps hauting my updates. I don't know why, but I know it's tied to the syntax of the decisions. I'm trying to figure out, but it is hard.

**Comment by derHochmeister** [comment]
The allow_title and owned_by_title are no news. I'm trying to understand why do they happen.

**Comment by derHochmeister** [comment]
It seems the first Philippine decision was giving some trouble, as well as some in Africa. I fixed them, while also correcting the renaming decisions, adding more flavor. I added renaming decisions for NYC and New Orléans for Scandinavians and Iberians, since they lack and they shouldn't, and their localisations will arrive in a later release.

Here is the pastebin (`decisions/converterDecisions.txt`): https://pastebin.com/HM8J8tBX

**Comment by kratostatic** [comment]
Brilliant, thank you again!

There's an AI_will_do decision now appearing but I'll get to work squashing that.  I've actually split the converter decisions file into three as it was getting quite unwieldy.  Oh and I added flags to the NYC/New Orleans province renaming so they only happen once.

**Comment by derHochmeister** [comment]
You still get an AI_will_do decision spam? Weird, I removed all the buggy things. Anyway, as soon as it is in the converter, it is ok. Also, I agree with splitting the decisions between "postcol_", nation formation, province-renaming.

As soon as it is implemented in the code, I think the release should be done, as the milestone is complete and all 1.22 is added.

**Comment by kratostatic** [comment]
Yeah I've just narrowed it down to the NYC renaming decisions, everything else is fine.

One other change I was thinking of making to the main post-colonial decision is only add the cores if the nation doesn't already exist so that preexisting small nations don't get a bunch of free cores for no reason (like that Dutch USA or in this game a Swedish Mexico).

And yeah, then release!

**Comment by derHochmeister** [comment]
For a next release, I'll narrow down this, as well as adding a "is_independant" to the post-colonial decisions, to avoid Scottish Venezuela forming Colombia and becoming independent with no reason.

**Comment by kratostatic** [comment]
And I've found the problem - no close bracket after New Aragon.

And looking at the decision syntax I don't think I can quickly change them to allow for only adding cores if a nation doesn't already exist.  I'll get on with releasing then!
